// react-native-purchases-ui/lib/commonjs/customVariables.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CustomVariableValue = void 0;
exports.convertCustomVariablesToStringMap = convertCustomVariablesToStringMap;
exports.transformOptionsForNative = transformOptionsForNative;
/**
 * A value type for custom paywall variables that can be passed to paywalls at runtime.
 *
 * Custom variables allow developers to personalize paywall text with dynamic values.
 * Variables are defined in the RevenueCat dashboard and can be overridden at runtime.
 *
 * Currently only string values are supported. Additional types may be added in the future.
 *
 * @example
 * ```typescript
 * RevenueCatUI.presentPaywall({
 *   customVariables: {
 *     'player_name': CustomVariableValue.string('John'),
 *     'level': CustomVariableValue.string('42'),
 *   },
 * });
 * ```
 *
 * In the paywall text (configured in the dashboard), use the `custom.` prefix:
 * ```
 * Hello {{ custom.player_name }}!
 * ```
 */

/**
 * Factory methods for creating CustomVariableValue instances.
 */
const CustomVariableValue = exports.CustomVariableValue = {
  /**
   * Creates a string custom variable value.
   * @param value The string value for the custom variable.
   * @returns A CustomVariableValue containing the string.
   */
  string: value => ({
    type: 'string',
    value
  })
};

/**
 * A map of custom variable names to their values.
 */

/**
 * Internal type for custom variables as sent to native bridge.
 * Currently only string values are supported.
 * @internal
 */

/**
 * Converts CustomVariables to a string map for native bridge.
 * @internal
 * @visibleForTesting
 */
function convertCustomVariablesToStringMap(customVariables) {
  if (!customVariables) return null;
  const result = {};
  for (const key of Object.keys(customVariables)) {
    const variable = customVariables[key];
    if (variable) {
      result[key] = variable.value;
    }
  }
  return result;
}

/**
 * Transforms options to native format, converting CustomVariables to string map.
 * @internal
 * @visibleForTesting
 */
function transformOptionsForNative(options) {
  if (!options) return undefined;
  const {
    customVariables,
    ...rest
  } = options;
  return {
    ...rest,
    customVariables: convertCustomVariablesToStringMap(customVariables)
  };
}
