// react-native-health-link/lib/commonjs/types/units.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.unitToIosUnitMap = exports.androidWeightUnitMap = exports.androidHeightUnitMap = exports.androidBloodGlucoseUnitMap = exports.WeightUnit = exports.StepsUnit = exports.HeighUnit = exports.HeartRateUnit = exports.BloodGlucoseUnit = void 0;
var _reactNativeHealth = require("react-native-health");
let BloodGlucoseUnit = exports.BloodGlucoseUnit = /*#__PURE__*/function (BloodGlucoseUnit) {
  BloodGlucoseUnit["MgPerdL"] = "mgPerdL";
  BloodGlucoseUnit["MmolPerL"] = "mmolPerL";
  return BloodGlucoseUnit;
}({});
let WeightUnit = exports.WeightUnit = /*#__PURE__*/function (WeightUnit) {
  WeightUnit["Pounds"] = "pounds";
  WeightUnit["Kg"] = "kg";
  WeightUnit["Gram"] = "gram";
  return WeightUnit;
}({});
let HeighUnit = exports.HeighUnit = /*#__PURE__*/function (HeighUnit) {
  HeighUnit["Inch"] = "inch";
  HeighUnit["Cm"] = "cm";
  HeighUnit["Meter"] = "meter";
  HeighUnit["Foot"] = "foot";
  return HeighUnit;
}({});
let StepsUnit = exports.StepsUnit = /*#__PURE__*/function (StepsUnit) {
  StepsUnit["Count"] = "count";
  return StepsUnit;
}({});
let HeartRateUnit = exports.HeartRateUnit = /*#__PURE__*/function (HeartRateUnit) {
  HeartRateUnit["Bpm"] = "bpm";
  return HeartRateUnit;
}({});
const androidHeightUnitMap = (data, unit) => {
  switch (unit) {
    case HeighUnit.Cm:
      return data.height.inMeters * 100;
    case HeighUnit.Meter:
      return data.height.inMeters;
    case HeighUnit.Foot:
      return data.height.inFeet;
    case HeighUnit.Inch:
      return data.height.inInches;
    default:
      return data.height.inInches;
  }
};
exports.androidHeightUnitMap = androidHeightUnitMap;
const androidWeightUnitMap = (data, unit) => {
  switch (unit) {
    case WeightUnit.Kg:
      return data.weight.inKilograms;
    case WeightUnit.Gram:
      return data.weight.inGrams;
    case WeightUnit.Pounds:
      return data.weight.inPounds;
    default:
      return data.weight.inPounds;
  }
};
exports.androidWeightUnitMap = androidWeightUnitMap;
const androidBloodGlucoseUnitMap = exports.androidBloodGlucoseUnitMap = {
  [BloodGlucoseUnit.MgPerdL]: 'milligramsPerDeciliter',
  [BloodGlucoseUnit.MmolPerL]: 'millimolesPerLiter'
};
const unitToIosUnitMap = exports.unitToIosUnitMap = {
  [HeighUnit.Meter]: _reactNativeHealth.HealthUnit?.meter,
  [HeighUnit.Foot]: _reactNativeHealth.HealthUnit?.foot,
  [HeighUnit.Inch]: _reactNativeHealth.HealthUnit?.inch,
  [WeightUnit.Gram]: _reactNativeHealth.HealthUnit?.gram,
  [WeightUnit.Pounds]: _reactNativeHealth.HealthUnit?.pound,
  [StepsUnit.Count]: _reactNativeHealth.HealthUnit?.count,
  [BloodGlucoseUnit.MgPerdL]: _reactNativeHealth.HealthUnit?.mgPerdL,
  [BloodGlucoseUnit.MmolPerL]: _reactNativeHealth.HealthUnit?.mmolPerL,
  [HeighUnit.Cm]: _reactNativeHealth.HealthUnit?.meter,
  [WeightUnit.Kg]: _reactNativeHealth.HealthUnit?.gram,
  [HeartRateUnit.Bpm]: _reactNativeHealth.HealthUnit?.bpm
};
