// react-native-health-link/lib/module/types/dataTypes.js
"use strict";

/**
 * Enum representing various types of health data that can be tracked.
 *
 * @enum {string}
 * @property {string} BloodGlucose - Represents blood glucose levels.
 * @property {string} Height - Represents height measurement.
 * @property {string} Weight - Represents weight measurement.
 * @property {string} HeartRate - Represents heart rate measurement.
 * @property {string} RestingHeartRate - Represents resting heart rate measurement.
 * @property {string} BloodPressure - Represents blood pressure measurement.
 * @property {string} OxygenSaturation - Represents oxygen saturation levels.
 * @property {string} Steps - Represents the number of steps taken.
 */
export let HealthLinkDataType = /*#__PURE__*/function (HealthLinkDataType) {
  HealthLinkDataType["BloodGlucose"] = "BloodGlucose";
  HealthLinkDataType["Height"] = "Height";
  HealthLinkDataType["Weight"] = "Weight";
  HealthLinkDataType["HeartRate"] = "HeartRate";
  HealthLinkDataType["RestingHeartRate"] = "RestingHeartRate";
  HealthLinkDataType["BloodPressure"] = "BloodPressure";
  HealthLinkDataType["OxygenSaturation"] = "OxygenSaturation";
  HealthLinkDataType["Steps"] = "Steps";
  return HealthLinkDataType;
}({});
export const optionsToAndroidOptions = options => {
  let operator = 'before';
  if (options.startDate && options.endDate) {
    operator = 'between';
  } else if (options.startDate) {
    operator = 'after';
  }
  return {
    timeRangeFilter: {
      operator,
      startTime: options.startDate,
      endTime: options.endDate
    },
    ascendingOrder: options.ascending ?? false,
    pageSize: options.limit
  };
};
