// react-native-health-link/lib/module/types/units.js
"use strict";

import { HealthUnit } from 'react-native-health';
export let BloodGlucoseUnit = /*#__PURE__*/function (BloodGlucoseUnit) {
  BloodGlucoseUnit["MgPerdL"] = "mgPerdL";
  BloodGlucoseUnit["MmolPerL"] = "mmolPerL";
  return BloodGlucoseUnit;
}({});
export let WeightUnit = /*#__PURE__*/function (WeightUnit) {
  WeightUnit["Pounds"] = "pounds";
  WeightUnit["Kg"] = "kg";
  WeightUnit["Gram"] = "gram";
  return WeightUnit;
}({});
export let HeighUnit = /*#__PURE__*/function (HeighUnit) {
  HeighUnit["Inch"] = "inch";
  HeighUnit["Cm"] = "cm";
  HeighUnit["Meter"] = "meter";
  HeighUnit["Foot"] = "foot";
  return HeighUnit;
}({});
export let StepsUnit = /*#__PURE__*/function (StepsUnit) {
  StepsUnit["Count"] = "count";
  return StepsUnit;
}({});
export let HeartRateUnit = /*#__PURE__*/function (HeartRateUnit) {
  HeartRateUnit["Bpm"] = "bpm";
  return HeartRateUnit;
}({});
export const androidHeightUnitMap = (data, unit) => {
  switch (unit) {
    case HeighUnit.Cm:
      return data.height.inMeters * 100;
    case HeighUnit.Meter:
      return data.height.inMeters;
    case HeighUnit.Foot:
      return data.height.inFeet;
    case HeighUnit.Inch:
      return data.height.inInches;
    default:
      return data.height.inInches;
  }
};
export const androidWeightUnitMap = (data, unit) => {
  switch (unit) {
    case WeightUnit.Kg:
      return data.weight.inKilograms;
    case WeightUnit.Gram:
      return data.weight.inGrams;
    case WeightUnit.Pounds:
      return data.weight.inPounds;
    default:
      return data.weight.inPounds;
  }
};
export const androidBloodGlucoseUnitMap = {
  [BloodGlucoseUnit.MgPerdL]: 'milligramsPerDeciliter',
  [BloodGlucoseUnit.MmolPerL]: 'millimolesPerLiter'
};
export const unitToIosUnitMap = {
  [HeighUnit.Meter]: HealthUnit?.meter,
  [HeighUnit.Foot]: HealthUnit?.foot,
  [HeighUnit.Inch]: HealthUnit?.inch,
  [WeightUnit.Gram]: HealthUnit?.gram,
  [WeightUnit.Pounds]: HealthUnit?.pound,
  [StepsUnit.Count]: HealthUnit?.count,
  [BloodGlucoseUnit.MgPerdL]: HealthUnit?.mgPerdL,
  [BloodGlucoseUnit.MmolPerL]: HealthUnit?.mmolPerL,
  [HeighUnit.Cm]: HealthUnit?.meter,
  [WeightUnit.Kg]: HealthUnit?.gram,
  [HeartRateUnit.Bpm]: HealthUnit?.bpm
};
