// react-native-health-link/lib/commonjs/helpers/permissions.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.genericToIosPermissions = exports.genericToAndroidPermissions = void 0;
var _reactNativeHealth = _interopRequireDefault(require("react-native-health"));
var _permissions = require("../types/permissions.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const genericToIosPermissions = permissions => {
  const readPermissions = permissions.read.reduce((acc, permission) => {
    if (Array.isArray(genericToIosMap[permission])) {
      acc.push(...genericToIosMap[permission]);
    }
    if (genericToIosMap[permission]) {
      acc.push(genericToIosMap[permission]);
    }
    return acc;
  }, []);
  const writePermissions = permissions.write.reduce((acc, permission) => {
    if (genericToIosMap[permission]) {
      acc.push(genericToIosMap[permission]);
    }
    return acc;
  }, []);
  return {
    permissions: {
      read: readPermissions.flat(),
      write: writePermissions.flat()
    }
  };
};
exports.genericToIosPermissions = genericToIosPermissions;
const genericToAndroidPermissions = permissions => {
  let androidPermissions = [];
  permissions.read.forEach(permission => {
    if (genericToAndroidMap[permission]) {
      androidPermissions.push({
        accessType: 'read',
        recordType: genericToAndroidMap[permission]
      });
    }
  });
  permissions.write.forEach(permission => {
    if (genericToAndroidMap[permission]) {
      androidPermissions.push({
        accessType: 'write',
        recordType: genericToAndroidMap[permission]
      });
    }
  });
  return androidPermissions;
};
exports.genericToAndroidPermissions = genericToAndroidPermissions;
const genericToIosMap = {
  [_permissions.HealthLinkPermissions.BloodGlucose]: _reactNativeHealth.default.Constants.Permissions.BloodGlucose,
  [_permissions.HealthLinkPermissions.Height]: _reactNativeHealth.default.Constants.Permissions.Height,
  [_permissions.HealthLinkPermissions.Weight]: _reactNativeHealth.default.Constants.Permissions.Weight,
  [_permissions.HealthLinkPermissions.HeartRate]: _reactNativeHealth.default.Constants.Permissions.HeartRate,
  [_permissions.HealthLinkPermissions.RestingHeartRate]: _reactNativeHealth.default.Constants.Permissions.RestingHeartRate,
  [_permissions.HealthLinkPermissions.BloodPressure]: [_reactNativeHealth.default.Constants.Permissions.BloodPressureDiastolic, _reactNativeHealth.default.Constants.Permissions.BloodPressureSystolic],
  [_permissions.HealthLinkPermissions.OxygenSaturation]: _reactNativeHealth.default.Constants.Permissions.OxygenSaturation,
  [_permissions.HealthLinkPermissions.Steps]: _reactNativeHealth.default.Constants.Permissions.Steps
};
const genericToAndroidMap = {
  [_permissions.HealthLinkPermissions.BloodGlucose]: 'BloodGlucose',
  [_permissions.HealthLinkPermissions.Height]: 'Height',
  [_permissions.HealthLinkPermissions.Weight]: 'Weight',
  [_permissions.HealthLinkPermissions.HeartRate]: 'HeartRate',
  [_permissions.HealthLinkPermissions.RestingHeartRate]: 'RestingHeartRate',
  [_permissions.HealthLinkPermissions.BloodPressure]: 'BloodPressure',
  [_permissions.HealthLinkPermissions.OxygenSaturation]: 'OxygenSaturation',
  [_permissions.HealthLinkPermissions.Steps]: 'Steps'
};
