// react-native-health-link/lib/commonjs/helpers/results.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.readIosCallback = exports.readDataResultDeserializer = exports.dataValueToIosReadFunction = exports.dataValueDeserializer = void 0;
var _reactNative = require("react-native");
var _dataTypes = require("../types/dataTypes.js");
var _units = require("../types/units.js");
const AppleHealthKit = require('react-native-health');
const dataValueDeserializer = (dataType, options, dataValue) => {
  if (_reactNative.Platform.OS === 'ios') {
    let iosDataValue = dataValue;
    let result = {
      value: iosDataValue.value,
      id: iosDataValue.id,
      time: iosDataValue.startDate,
      metadata: {
        ...iosDataValue.metadata,
        source: iosDataValue.sourceId
      }
    };
    switch (dataType) {
      case _dataTypes.HealthLinkDataType.BloodPressure:
        return {
          ...result,
          value: {
            systolic: iosDataValue.bloodPressureSystolicValue,
            diastolic: iosDataValue.bloodPressureDiastolicValue
          }
        };
      case _dataTypes.HealthLinkDataType.OxygenSaturation:
        return {
          ...result,
          value: iosDataValue.value * 100
        };
      default:
        return result;
    }
  } else if (_reactNative.Platform.OS === 'android') {
    let androidDataValue = dataValue;
    let result = {
      id: androidDataValue.metadata?.id,
      metadata: {
        ...androidDataValue.metadata,
        source: androidDataValue.metadata?.dataOrigin
      },
      time: androidDataValue.time
    };
    switch (dataType) {
      case _dataTypes.HealthLinkDataType.BloodGlucose:
        {
          const bloodGlucoseDataValue = androidDataValue;
          return {
            ...result,
            metadata: {
              ...result.metadata,
              relationToMeal: bloodGlucoseDataValue.relationToMeal,
              mealType: bloodGlucoseDataValue.mealType,
              specimenSource: bloodGlucoseDataValue.specimenSource
            },
            value: options?.unit === _units.BloodGlucoseUnit.MgPerdL ? bloodGlucoseDataValue.level?.inMilligramsPerDeciliter : bloodGlucoseDataValue.level?.inMillimolesPerLiter
          };
        }
      case _dataTypes.HealthLinkDataType.BloodPressure:
        {
          const bloodPressureDataValue = androidDataValue;
          return {
            ...result,
            value: {
              systolic: bloodPressureDataValue.systolic.inMillimetersOfMercury,
              diastolic: bloodPressureDataValue.diastolic.inMillimetersOfMercury
            }
          };
        }
      case _dataTypes.HealthLinkDataType.Height:
        {
          const heightDataValue = androidDataValue;
          return {
            ...result,
            value: (0, _units.androidHeightUnitMap)(heightDataValue, options.unit)
          };
        }
      case _dataTypes.HealthLinkDataType.Weight:
        {
          const weightDataValue = androidDataValue;
          return {
            ...result,
            value: (0, _units.androidWeightUnitMap)(weightDataValue, options.unit)
          };
        }
      case _dataTypes.HealthLinkDataType.HeartRate:
        {
          const heartRateDataValue = androidDataValue;
          return {
            ...result,
            value: heartRateDataValue.samples[0]?.beatsPerMinute,
            time: heartRateDataValue.samples[0]?.time
          };
        }
      case _dataTypes.HealthLinkDataType.RestingHeartRate:
        {
          const restingHeartRateDataValue = androidDataValue;
          return {
            ...result,
            value: restingHeartRateDataValue.beatsPerMinute
          };
        }
      case _dataTypes.HealthLinkDataType.OxygenSaturation:
        {
          const oxygenSaturationDataValue = androidDataValue;
          return {
            ...result,
            value: oxygenSaturationDataValue.percentage
          };
        }
      case _dataTypes.HealthLinkDataType.Steps:
        {
          const stepsDataValue = androidDataValue;
          return {
            ...result,
            value: stepsDataValue.count
          };
        }
      default:
        return result;
    }
  }
  return null;
};
exports.dataValueDeserializer = dataValueDeserializer;
const readDataResultDeserializer = (dataType, options, data) => {
  let dataValueArray = _reactNative.Platform.OS === 'ios' ? data : data.records;
  return dataValueArray.reduce((acc, d) => {
    const deserializedValue = dataValueDeserializer(dataType, options, d);
    if (deserializedValue !== null) {
      acc.push(deserializedValue);
    }
    return acc;
  }, []);
};
exports.readDataResultDeserializer = readDataResultDeserializer;
const readIosCallback = (dataType, options) => {
  return new Promise((resolve, reject) => {
    dataValueToIosReadFunction(dataType)(options, (err, results) => {
      if (err) {
        console.error(err);
        reject(err);
      }
      resolve(results);
    });
  });
};
exports.readIosCallback = readIosCallback;
const dataValueToIosReadFunction = dataType => {
  const dataTypeMap = {
    BloodGlucose: AppleHealthKit.getBloodGlucoseSamples,
    Height: AppleHealthKit.getHeightSamples,
    Weight: AppleHealthKit.getWeightSamples,
    HeartRate: AppleHealthKit.getHeartRateSamples,
    RestingHeartRate: AppleHealthKit.getRestingHeartRateSamples,
    OxygenSaturation: AppleHealthKit.getOxygenSaturationSamples,
    BloodPressure: AppleHealthKit.getBloodPressureSamples,
    Steps: AppleHealthKit.getDailyStepCountSamples
  };
  return dataTypeMap[dataType] || (() => {});
};
exports.dataValueToIosReadFunction = dataValueToIosReadFunction;
