// react-native-health-link/lib/commonjs/helpers/save.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.writeIosCallback = exports.serializeWriteOptions = exports.dataValueToIosWriteFunction = void 0;
var _dataTypes = require("../types/dataTypes.js");
var _reactNative = require("react-native");
var _units = require("../types/units.js");
const AppleHealthKit = require('react-native-health');
const serializeWriteOptions = (dataType, options) => {
  if (_reactNative.Platform.OS === 'ios') {
    if (!options.unit) return null;
    const value = options.value ?? 0;
    const iosOptions = {
      unit: _units.unitToIosUnitMap[options.unit],
      value: options.unit === _units.WeightUnit.Kg ? value * 1000 : options.unit === _units.HeighUnit.Cm ? value / 100 : value
    };
    return iosOptions;
  }
  if (_reactNative.Platform.OS === 'android') {
    const androidOptions = {
      startTime: options.startDate ?? new Date().toISOString(),
      endTime: options.endDate ?? new Date().toISOString()
    };
    switch (dataType) {
      case _dataTypes.HealthLinkDataType.BloodGlucose:
        {
          const {
            relationToMeal = 0,
            mealType = 0,
            specimenSource = 0
          } = options.metadata || {};
          const unit = options.unit === _units.BloodGlucoseUnit.MgPerdL ? 'milligramsPerDeciliter' : 'millimolesPerLiter';
          return {
            ...androidOptions,
            recordType: 'BloodGlucose',
            relationToMeal,
            mealType,
            specimenSource,
            time: options.time ?? androidOptions.startTime,
            level: {
              unit,
              value: options.value
            }
          };
        }
      case _dataTypes.HealthLinkDataType.Steps:
        return {
          ...androidOptions,
          recordType: 'Steps',
          count: options.value ?? 0
        };
      case _dataTypes.HealthLinkDataType.Weight:
        {
          const weightUnit = options.unit === _units.WeightUnit.Kg ? 'kilograms' : options.unit === _units.WeightUnit.Gram ? 'grams' : 'pounds';
          return {
            ...androidOptions,
            recordType: 'Weight',
            time: options.time ?? androidOptions.startTime,
            weight: {
              unit: weightUnit,
              value: options.value
            }
          };
        }
      case _dataTypes.HealthLinkDataType.Height:
        {
          const heightUnit = options.unit === _units.HeighUnit.Cm || options.unit === _units.HeighUnit.Meter ? 'meters' : options.unit === _units.HeighUnit.Foot ? 'feet' : 'inches';
          const heightValue = options.unit === _units.HeighUnit.Cm ? options.value / 100 : options.value;
          return {
            ...androidOptions,
            recordType: 'Height',
            time: options.time ?? androidOptions.startTime,
            height: {
              unit: heightUnit,
              value: heightValue
            }
          };
        }
      case _dataTypes.HealthLinkDataType.HeartRate:
        return {
          ...androidOptions,
          recordType: 'HeartRate',
          samples: [{
            time: options.time ?? androidOptions.startTime,
            beatsPerMinute: options.value
          }]
        };
      default:
        return null;
    }
  }
  return null;
};
exports.serializeWriteOptions = serializeWriteOptions;
const writeIosCallback = (dataType, options) => {
  return new Promise((resolve, reject) => {
    dataValueToIosWriteFunction(dataType)(options, (err, results) => {
      if (err) {
        console.error(err);
        reject(err);
      }
      resolve(results);
    });
  });
};
exports.writeIosCallback = writeIosCallback;
const dataValueToIosWriteFunction = dataType => {
  const dataTypeMap = {
    BloodGlucose: AppleHealthKit.saveBloodGlucoseSample,
    Height: AppleHealthKit.saveHeight,
    Weight: AppleHealthKit.saveWeight,
    HeartRate: AppleHealthKit.saveHeartRateSample,
    Steps: AppleHealthKit.saveSteps
  };
  return dataTypeMap[dataType] || (() => {});
};
exports.dataValueToIosWriteFunction = dataValueToIosWriteFunction;
