// react-native-health-link/lib/module/helpers/permissions.js
"use strict";

import AppleHealthKit from 'react-native-health';
import { HealthLinkPermissions } from "../types/permissions.js";
export const genericToIosPermissions = permissions => {
  const readPermissions = permissions.read.reduce((acc, permission) => {
    if (Array.isArray(genericToIosMap[permission])) {
      acc.push(...genericToIosMap[permission]);
    }
    if (genericToIosMap[permission]) {
      acc.push(genericToIosMap[permission]);
    }
    return acc;
  }, []);
  const writePermissions = permissions.write.reduce((acc, permission) => {
    if (genericToIosMap[permission]) {
      acc.push(genericToIosMap[permission]);
    }
    return acc;
  }, []);
  return {
    permissions: {
      read: readPermissions.flat(),
      write: writePermissions.flat()
    }
  };
};
export const genericToAndroidPermissions = permissions => {
  let androidPermissions = [];
  permissions.read.forEach(permission => {
    if (genericToAndroidMap[permission]) {
      androidPermissions.push({
        accessType: 'read',
        recordType: genericToAndroidMap[permission]
      });
    }
  });
  permissions.write.forEach(permission => {
    if (genericToAndroidMap[permission]) {
      androidPermissions.push({
        accessType: 'write',
        recordType: genericToAndroidMap[permission]
      });
    }
  });
  return androidPermissions;
};
const genericToIosMap = {
  [HealthLinkPermissions.BloodGlucose]: AppleHealthKit.Constants.Permissions.BloodGlucose,
  [HealthLinkPermissions.Height]: AppleHealthKit.Constants.Permissions.Height,
  [HealthLinkPermissions.Weight]: AppleHealthKit.Constants.Permissions.Weight,
  [HealthLinkPermissions.HeartRate]: AppleHealthKit.Constants.Permissions.HeartRate,
  [HealthLinkPermissions.RestingHeartRate]: AppleHealthKit.Constants.Permissions.RestingHeartRate,
  [HealthLinkPermissions.BloodPressure]: [AppleHealthKit.Constants.Permissions.BloodPressureDiastolic, AppleHealthKit.Constants.Permissions.BloodPressureSystolic],
  [HealthLinkPermissions.OxygenSaturation]: AppleHealthKit.Constants.Permissions.OxygenSaturation,
  [HealthLinkPermissions.Steps]: AppleHealthKit.Constants.Permissions.Steps
};
const genericToAndroidMap = {
  [HealthLinkPermissions.BloodGlucose]: 'BloodGlucose',
  [HealthLinkPermissions.Height]: 'Height',
  [HealthLinkPermissions.Weight]: 'Weight',
  [HealthLinkPermissions.HeartRate]: 'HeartRate',
  [HealthLinkPermissions.RestingHeartRate]: 'RestingHeartRate',
  [HealthLinkPermissions.BloodPressure]: 'BloodPressure',
  [HealthLinkPermissions.OxygenSaturation]: 'OxygenSaturation',
  [HealthLinkPermissions.Steps]: 'Steps'
};
