// react-native-health-link/lib/module/helpers/results.js
"use strict";

import { Platform } from 'react-native';
import { HealthLinkDataType } from "../types/dataTypes.js";
import { androidHeightUnitMap, androidWeightUnitMap, BloodGlucoseUnit } from "../types/units.js";
const AppleHealthKit = require('react-native-health');
export const dataValueDeserializer = (dataType, options, dataValue) => {
  if (Platform.OS === 'ios') {
    let iosDataValue = dataValue;
    let result = {
      value: iosDataValue.value,
      id: iosDataValue.id,
      time: iosDataValue.startDate,
      metadata: {
        ...iosDataValue.metadata,
        source: iosDataValue.sourceId
      }
    };
    switch (dataType) {
      case HealthLinkDataType.BloodPressure:
        return {
          ...result,
          value: {
            systolic: iosDataValue.bloodPressureSystolicValue,
            diastolic: iosDataValue.bloodPressureDiastolicValue
          }
        };
      case HealthLinkDataType.OxygenSaturation:
        return {
          ...result,
          value: iosDataValue.value * 100
        };
      default:
        return result;
    }
  } else if (Platform.OS === 'android') {
    let androidDataValue = dataValue;
    let result = {
      id: androidDataValue.metadata?.id,
      metadata: {
        ...androidDataValue.metadata,
        source: androidDataValue.metadata?.dataOrigin
      },
      time: androidDataValue.time
    };
    switch (dataType) {
      case HealthLinkDataType.BloodGlucose:
        {
          const bloodGlucoseDataValue = androidDataValue;
          return {
            ...result,
            metadata: {
              ...result.metadata,
              relationToMeal: bloodGlucoseDataValue.relationToMeal,
              mealType: bloodGlucoseDataValue.mealType,
              specimenSource: bloodGlucoseDataValue.specimenSource
            },
            value: options?.unit === BloodGlucoseUnit.MgPerdL ? bloodGlucoseDataValue.level?.inMilligramsPerDeciliter : bloodGlucoseDataValue.level?.inMillimolesPerLiter
          };
        }
      case HealthLinkDataType.BloodPressure:
        {
          const bloodPressureDataValue = androidDataValue;
          return {
            ...result,
            value: {
              systolic: bloodPressureDataValue.systolic.inMillimetersOfMercury,
              diastolic: bloodPressureDataValue.diastolic.inMillimetersOfMercury
            }
          };
        }
      case HealthLinkDataType.Height:
        {
          const heightDataValue = androidDataValue;
          return {
            ...result,
            value: androidHeightUnitMap(heightDataValue, options.unit)
          };
        }
      case HealthLinkDataType.Weight:
        {
          const weightDataValue = androidDataValue;
          return {
            ...result,
            value: androidWeightUnitMap(weightDataValue, options.unit)
          };
        }
      case HealthLinkDataType.HeartRate:
        {
          const heartRateDataValue = androidDataValue;
          return {
            ...result,
            value: heartRateDataValue.samples[0]?.beatsPerMinute,
            time: heartRateDataValue.samples[0]?.time
          };
        }
      case HealthLinkDataType.RestingHeartRate:
        {
          const restingHeartRateDataValue = androidDataValue;
          return {
            ...result,
            value: restingHeartRateDataValue.beatsPerMinute
          };
        }
      case HealthLinkDataType.OxygenSaturation:
        {
          const oxygenSaturationDataValue = androidDataValue;
          return {
            ...result,
            value: oxygenSaturationDataValue.percentage
          };
        }
      case HealthLinkDataType.Steps:
        {
          const stepsDataValue = androidDataValue;
          return {
            ...result,
            value: stepsDataValue.count
          };
        }
      default:
        return result;
    }
  }
  return null;
};
export const readDataResultDeserializer = (dataType, options, data) => {
  let dataValueArray = Platform.OS === 'ios' ? data : data.records;
  return dataValueArray.reduce((acc, d) => {
    const deserializedValue = dataValueDeserializer(dataType, options, d);
    if (deserializedValue !== null) {
      acc.push(deserializedValue);
    }
    return acc;
  }, []);
};
export const readIosCallback = (dataType, options) => {
  return new Promise((resolve, reject) => {
    dataValueToIosReadFunction(dataType)(options, (err, results) => {
      if (err) {
        console.error(err);
        reject(err);
      }
      resolve(results);
    });
  });
};
export const dataValueToIosReadFunction = dataType => {
  const dataTypeMap = {
    BloodGlucose: AppleHealthKit.getBloodGlucoseSamples,
    Height: AppleHealthKit.getHeightSamples,
    Weight: AppleHealthKit.getWeightSamples,
    HeartRate: AppleHealthKit.getHeartRateSamples,
    RestingHeartRate: AppleHealthKit.getRestingHeartRateSamples,
    OxygenSaturation: AppleHealthKit.getOxygenSaturationSamples,
    BloodPressure: AppleHealthKit.getBloodPressureSamples,
    Steps: AppleHealthKit.getDailyStepCountSamples
  };
  return dataTypeMap[dataType] || (() => {});
};
