// react-native-health-link/lib/module/helpers/save.js
"use strict";

import { HealthLinkDataType } from "../types/dataTypes.js";
import { Platform } from 'react-native';
import { BloodGlucoseUnit, HeighUnit, unitToIosUnitMap, WeightUnit } from "../types/units.js";
const AppleHealthKit = require('react-native-health');
export const serializeWriteOptions = (dataType, options) => {
  if (Platform.OS === 'ios') {
    if (!options.unit) return null;
    const value = options.value ?? 0;
    const iosOptions = {
      unit: unitToIosUnitMap[options.unit],
      value: options.unit === WeightUnit.Kg ? value * 1000 : options.unit === HeighUnit.Cm ? value / 100 : value
    };
    return iosOptions;
  }
  if (Platform.OS === 'android') {
    const androidOptions = {
      startTime: options.startDate ?? new Date().toISOString(),
      endTime: options.endDate ?? new Date().toISOString()
    };
    switch (dataType) {
      case HealthLinkDataType.BloodGlucose:
        {
          const {
            relationToMeal = 0,
            mealType = 0,
            specimenSource = 0
          } = options.metadata || {};
          const unit = options.unit === BloodGlucoseUnit.MgPerdL ? 'milligramsPerDeciliter' : 'millimolesPerLiter';
          return {
            ...androidOptions,
            recordType: 'BloodGlucose',
            relationToMeal,
            mealType,
            specimenSource,
            time: options.time ?? androidOptions.startTime,
            level: {
              unit,
              value: options.value
            }
          };
        }
      case HealthLinkDataType.Steps:
        return {
          ...androidOptions,
          recordType: 'Steps',
          count: options.value ?? 0
        };
      case HealthLinkDataType.Weight:
        {
          const weightUnit = options.unit === WeightUnit.Kg ? 'kilograms' : options.unit === WeightUnit.Gram ? 'grams' : 'pounds';
          return {
            ...androidOptions,
            recordType: 'Weight',
            time: options.time ?? androidOptions.startTime,
            weight: {
              unit: weightUnit,
              value: options.value
            }
          };
        }
      case HealthLinkDataType.Height:
        {
          const heightUnit = options.unit === HeighUnit.Cm || options.unit === HeighUnit.Meter ? 'meters' : options.unit === HeighUnit.Foot ? 'feet' : 'inches';
          const heightValue = options.unit === HeighUnit.Cm ? options.value / 100 : options.value;
          return {
            ...androidOptions,
            recordType: 'Height',
            time: options.time ?? androidOptions.startTime,
            height: {
              unit: heightUnit,
              value: heightValue
            }
          };
        }
      case HealthLinkDataType.HeartRate:
        return {
          ...androidOptions,
          recordType: 'HeartRate',
          samples: [{
            time: options.time ?? androidOptions.startTime,
            beatsPerMinute: options.value
          }]
        };
      default:
        return null;
    }
  }
  return null;
};
export const writeIosCallback = (dataType, options) => {
  return new Promise((resolve, reject) => {
    dataValueToIosWriteFunction(dataType)(options, (err, results) => {
      if (err) {
        console.error(err);
        reject(err);
      }
      resolve(results);
    });
  });
};
export const dataValueToIosWriteFunction = dataType => {
  const dataTypeMap = {
    BloodGlucose: AppleHealthKit.saveBloodGlucoseSample,
    Height: AppleHealthKit.saveHeight,
    Weight: AppleHealthKit.saveWeight,
    HeartRate: AppleHealthKit.saveHeartRateSample,
    Steps: AppleHealthKit.saveSteps
  };
  return dataTypeMap[dataType] || (() => {});
};
